plugins/spender/splice.c

```diff
@@ -0,0 +1,1402 @@
+					    "Should be no out_ppm on final");
+		if (splice_cmd->actions[i]->pays_fee && pays_fee++)
+			return do_fail(cmd, splice_cmd, JSONRPC2_INVALID_PARAMS,
+					    "Only one item may pay fee");
+		if (splice_cmd->actions[i]->channel_id)
+			dest_count++;
+		if (splice_cmd->actions[i]->bitcoin_address)
+			dest_count++;
+		if (splice_cmd->actions[i]->onchain_wallet)
+			dest_count++;
+		if (dest_count < 1)
+			return do_fail(cmd, splice_cmd, JSONRPC2_INVALID_PARAMS,
+					    "Must specify 1 destination per");
+		if (dest_count > 1)
+			return do_fail(cmd, splice_cmd, JSONRPC2_INVALID_PARAMS,
+					    "Too many destinations per");
+
+		/* If user specifies both sats in and out, we just use the
+		 * larger of the two and subtract the smaller. */
+		if (amount_sat_greater(action->in_sat, action->out_sat)) {
+			if (!amount_sat_sub(&action->in_sat, action->in_sat,
+					    action->out_sat))
+				return do_fail(cmd, splice_cmd,
+						    JSONRPC2_INVALID_PARAMS,
+						    "Unable to sub out_sat from"
+						    " in_sat");
+			action->out_sat = amount_sat(0);
+		} else {
+			if (!amount_sat_sub(&action->out_sat, action->out_sat,
+					    action->in_sat))
+				return do_fail(cmd, splice_cmd,
+						    JSONRPC2_INVALID_PARAMS,
+						    "Unable to sub in_sat from"
+						    " out_sat");
+			action->in_sat = amount_sat(0);
+		}
+	}
+
+	add_to_debug_log(splice_cmd, "execute_splice");
+
+	for (size_t i = 0; i < tal_count(splice_cmd->actions); i++) {
+		action = splice_cmd->actions[i];
+		state = splice_cmd->states[i];
+
+		/* Load (only one) feerate if user provided one */
+		if (action->feerate_per_kw) {
+			if (splice_cmd->feerate_per_kw)
+				return do_fail(cmd, splice_cmd,
+						    JSONRPC2_INVALID_PARAMS,
+						    "Only one item may set"
+						    " feerate");
+			splice_cmd->feerate_per_kw = action->feerate_per_kw;
+		}
+
+		/* Fund out to bitcoin address */
+		if (action->bitcoin_address) {
+			if (!amount_sat_zero(action->in_sat))
+				return do_fail(cmd, splice_cmd,
+						    JSONRPC2_INVALID_PARAMS,
+						    "Cannot fund from bitcoin"
+						    " address");
+			tok.type = JSMN_STRING;
+			tok.start = 0;
+			tok.end = strlen(action->bitcoin_address);
+			tok.size = tok.end;
+			switch(json_to_address_scriptpubkey(cmd,
+							    chainparams,
+							    action->bitcoin_address,
+							    &tok,
+							    &scriptpubkey)) {
+				case ADDRESS_PARSE_UNRECOGNIZED:
+					return do_fail(cmd, splice_cmd,
+							    JSONRPC2_INVALID_PARAMS,
+							    "Bitcoin address"
+							    " unrecognized");
+				case ADDRESS_PARSE_WRONG_NETWORK:
+					return do_fail(cmd, splice_cmd,
+							    JSONRPC2_INVALID_PARAMS,
+							    "Bitcoin address not on"
+							    " correct network");
+				case ADDRESS_PARSE_SUCCESS:
+					break;
+			}
+			output = psbt_append_output(splice_cmd->psbt,
+						    scriptpubkey,
+						    action->in_sat);
+
+			/* DTODO: should be action->in_sat but we haven't calculated those yet... */
```

Comment:
what does this mean??

Reply:
Updated to more meaningful `DTODO: support dynamic address payouts (percent)`